src/modules/powerrename/lib/PowerRenameManager.cpp

```diff
@@ -806,6 +806,19 @@ DWORD WINAPI CPowerRenameManager::s_regexWorkerThread(_In_ void* pv)
                                 {
                                     newNameToUse = nullptr;
                                 }
+                                
+
+                                wchar_t datedName[MAX_PATH] = { 0 };
+                                wchar_t date[MAX_PATH] = { 0 };
+                                SYSTEMTIME LocalTime;
+                                if (newNameToUse != nullptr)
+                                {
+                                    spItem->get_date(&LocalTime);
```

Comment:
When renaming a large amount of file, the cost of retrieving the date for all of them can be expensive, it would be better to do it only if necessary.
I would also consider using a different symbol instead of the `$` sign, even an invalid symbol for the filename for example (if there is any that would fit well).